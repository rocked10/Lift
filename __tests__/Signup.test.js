import React from 'react';
import { render, fireEvent, waitFor, act } from "@testing-library/react-native";
import renderer from 'react-test-renderer';

import Signup from "screens/signup";

describe('Sign up', () => {
    const wrapper = renderer.create(<Signup/>);

    it ('Sign up form submits correctly', async () => {
        const handleSubmit = jest.fn();
        const {getByTestId, getByPlaceholderText} = render(<Signup testing={true} handleSubmit={handleSubmit}/>);

        fireEvent.changeText(getByPlaceholderText('Enter your email...'), 'Test Email');
        fireEvent.changeText(getByPlaceholderText('Enter a username...'), 'Test Username');
        fireEvent.changeText(getByPlaceholderText('Enter a password...'), 'Test Password');
        fireEvent.changeText(getByPlaceholderText('Confirm your password...'), 'Test Password');

        fireEvent.press(getByTestId('Sign Up Button'));

        await waitFor(() => {
            expect(handleSubmit).toHaveBeenCalledWith(
                expect.objectContaining({
                    email: 'Test Email',
                    password: 'Test Password',
                    password2: 'Test Password',
                    role: 'Coach',
                    username: 'Test Username'
                }),
                expect.any(Object),
            );
        });
    });

    it ('Should render', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it ("Renders elements", () => {
        const navigate = jest.fn();
        const { getByTestId, getByText, getByPlaceholderText} = render(<Signup />);
        getByPlaceholderText('Enter your email...');
        getByPlaceholderText('Enter a username...');
        getByPlaceholderText('Enter a password...');
        getByPlaceholderText('Confirm your password...');
        getByText('I am a...');
        getByTestId('Role Picker')
        getByText('SIGNUP');
    });
});