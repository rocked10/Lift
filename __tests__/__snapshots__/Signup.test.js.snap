// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D3D3D3",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 20,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter your email..."
        rejectResponderTermination={true}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D3D3D3",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 20,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter a username..."
        rejectResponderTermination={true}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D3D3D3",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 20,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter a password..."
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D3D3D3",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 20,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Confirm your password..."
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "marginLeft": 8,
        }
      }
    >
      I am a...
    </Text>
    <View>
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Coach",
              "testID": undefined,
              "textColor": undefined,
              "value": "Coach",
            },
            Object {
              "label": "Athlete",
              "testID": undefined,
              "textColor": undefined,
              "value": "Athlete",
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="Role Picker"
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginVertical": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="Sign Up Button"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "karla-bold",
                }
              }
            >
              SIGNUP
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
